adapters/jwplayer/jwplayer_test.go

```diff
@@ -673,6 +735,54 @@ func TestSourceSanitization(t *testing.T) {
 	a := getTestAdapter()
 	var reqInfo adapters.ExtraRequestInfo
 
+	request := &openrtb2.BidRequest{
+		ID: "test_id",
+		Imp: []openrtb2.Imp{{
+			ID:  "test_imp_id",
+			Ext: json.RawMessage(`{"bidder":{"placementId": "1"}}`),
+		}},
+		Site: &openrtb2.Site{
+			Publisher: &openrtb2.Publisher{
+				Ext: json.RawMessage(`{"jwplayer":{"publisherId": "testPublisherId"}}`),
+			},
+		},
+		Source: &openrtb2.Source{
+			Ext: json.RawMessage(`{}`),
+		},
+	}
+
+	fmt.Printf("TestSourceSanitization")
+
+	processedRequests, err := a.MakeRequests(request, &reqInfo)
+	assert.Empty(t, err)
+
+	processedRequest := processedRequests[0]
+	bidRequest := &openrtb2.BidRequest{}
+	json.Unmarshal(processedRequest.Body, bidRequest)
+
+	expectedRequest := &openrtb2.BidRequest{
+		ID: "test_id",
+		Imp: []openrtb2.Imp{{
+			ID:    "test_imp_id",
+			Video: &openrtb2.Video{},
+			Ext:   json.RawMessage(`{"appnexus":{"placement_id":1}}`),
+		}},
+		Site: &openrtb2.Site{
+			Publisher: &openrtb2.Publisher{
+				Ext: json.RawMessage((`{"jwplayer":{"publisherId":"testPublisherId"}}`)),
+			},
+		},
+		Source: &openrtb2.Source{},
+		Device: &openrtb2.Device{},
+		Ext:    json.RawMessage((`{"schain":{"complete":1,"nodes":[{"asi":"jwplayer.com","sid":"testPublisherId","rid":"test_id","hp":1}],"ver":"1.0"}}`)),
+	}
+	assert.Equal(t, expectedRequest, bidRequest)
+}
+
+func TestSourceSanitization2(t *testing.T) {
```

Comment:
it seems like you don't understand the point of this unit test, because your test is not sanitizing anyting.
What you want to see is that the 2.6 schain gets removed. So your request has to have a 2.6 schain. The sanitized request should have it removed and should only have a 2.4 schain. 
That's the reason why after parsing the 2.6 schain there is a call to `source.SChain = nil`

Reply:
fixed.